packages/@lwc/engine/src/framework/wiring.ts

```diff
@@ -104,7 +102,14 @@ function createContextWatcher(
 }
 
 function createConnector(vm: VM, name: string, wireDef: WireDef): WireAdapter {
-    const { method, adapter } = wireDef;
+    const { method } = wireDef;
```

Comment:
what is the reason for moving this here instead of storeWireMethodMeta?

Reply:
@caridy `storeWireMethodMeta` happens while doing `registerDecorators`,  the issue is that in the test (and most of the places were we add a wire adapter using `register`) `registerDecorators` is called before the call to register the wireAdapter.

This change, basically stores the adapterId (the reference) in `storeWireMethodMeta`, and when creating the component and installing the wires, we check if it has the `.adapter` added by the register call and use it.